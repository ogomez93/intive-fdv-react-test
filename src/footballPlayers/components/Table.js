import React from 'react';
import { connect } from 'react-redux';
import VisiblePlayersList from './VisiblePlayersList';
import './css/table.css';

const Table = () => (
  <table id="playersTable" className="custom-table w-full text-center text-xs border">
    <thead>
      <tr label="labels" className="text-white font-semibold">
        <td label="Player" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Player</td>
        <td label="Position" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Position</td>
        <td label="Team" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Team</td>
        <td label="Age" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Age</td>
      </tr>
    </thead>
    <VisiblePlayersList />
  </table>
);

export default connect()(Table);