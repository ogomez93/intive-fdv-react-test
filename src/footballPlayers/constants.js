export const NAME = 'footballPlayers';
export const PLAYERS_API = 'https://football-players-b31f2.firebaseio.com/players.json?print=pretty';