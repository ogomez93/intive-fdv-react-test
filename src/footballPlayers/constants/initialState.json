{
  "fetching": true,
  "players": [],
  "nameFilter": "",
  "ageFilter": "",
  "positionFilter": ""
}
