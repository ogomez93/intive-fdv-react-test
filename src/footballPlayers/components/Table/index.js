import { branch, compose, renderComponent } from 'recompose';
import { connect } from 'react-redux';

import Table from './Table';
import Loader from '../Loader';

const mapStateToProps = state => ({
  fetching: state.footballPlayers.fetching
});

export default compose(
  connect(mapStateToProps),
  branch(
    ({ fetching }) => fetching,
    renderComponent(Loader)
  )
)(Table);
