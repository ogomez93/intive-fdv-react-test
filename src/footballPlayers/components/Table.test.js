import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import Table from './Table';
import { initialState } from './testConstants/Table.constants';

const mockStore = configureStore();

const store = mockStore(initialState);

describe('Component: <Table />', () => {
  const wrapper = shallow(<Table store={store} />);
  const component = wrapper.dive();
  const componentToJson = toJson(component);

  test('renders the component', () => {
    expect(componentToJson).toMatchSnapshot();
  });
  
  test('renders a table', () => {
    expect(componentToJson.type).toEqual('table');
  });

  const thead = componentToJson.children.find((el) => el.type === 'thead');
  const tr = thead.children[0];

  test('renders the Player table header', () => {
    const playerTd = tr.children.find((el) => el.props.label === 'Player');
    expect(playerTd.children[0]).toEqual('Player');
  });

  test('renders the Position table header', () => {
    const positionTd = tr.children.find((el) => el.props.label === 'Position');
    expect(positionTd.children[0]).toEqual('Position');
  });

  test('renders the Team table header', () => {
    const teamTd = tr.children.find((el) => el.props.label === 'Team');
    expect(teamTd.children[0]).toEqual('Team');
  });

  test('renders the Age table header', () => {
    const ageTd = tr.children.find((el) => el.props.label === 'Age');
    expect(ageTd.children[0]).toEqual('Age');
  });

  test('renders the VisiblePlayersList component', () => {
    const visiblePlayersList = componentToJson.children.find((el) => el.type === 'Connect(VisiblePlayersList');
    expect(visiblePlayersList).not.toBeNull();
  });
});
