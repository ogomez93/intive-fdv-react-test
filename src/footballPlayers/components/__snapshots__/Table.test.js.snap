// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Table /> renders the component 1`] = `
<table
  className="custom-table w-full text-center text-xs border"
  id="playersTable"
>
  <thead>
    <tr
      className="text-white font-semibold"
      label="labels"
    >
      <td
        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
        label="Player"
      >
        Player
      </td>
      <td
        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
        label="Position"
      >
        Position
      </td>
      <td
        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
        label="Team"
      >
        Team
      </td>
      <td
        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
        label="Age"
      >
        Age
      </td>
    </tr>
  </thead>
  <Connect(VisiblePlayersList) />
</table>
`;
